docs-generation/scripts/batch_translate_docs.py

```
@@ -330,6 +334,52 @@ def process_node(node_name, target_lang, lang_config, api_key, base_url, model,
+    aborted = False
+
+    with ThreadPoolExecutor(max_workers=concurrency) as pool:
+        # Submit in order, consume in completion order
+        future_to_node = {pool.submit(process_fn, n): n for n in nodes}
+        for fut in as_completed(future_to_node):
+            node_name = future_to_node[fut]
+            if fut.cancelled():
+                continue  # cancelled by the circuit breaker; not tallied
+            try:
+                result = fut.result()
+            except Exception as e:  # defensive; process_node already catches
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)